### PR TITLE
B: SCA warning fixes

### PR DESCRIPTION
**Pull request by MrPepperoni**

### CODE REVIEW

File: src/core/free_for_all_game_mode_system.cpp [hunk 1]
```diff
@@ -104,6 +104,7 @@ void FreeForAllGameModeSystem::OnStartGameMode( core::StartGameModeEvent const&
     {
         return;
     }
+#if 0
```
Comment (weanti) [inline src/core/free_for_all_game_mode_system.cpp]: do you plan to keep the code or use in the near future? if no, then please don't use #if 0

Comment (MrPepperoni) [inline src/core/free_for_all_game_mode_system.cpp]: nope, good point.

File: src/platform/tgatexture.cpp [hunk 2]
```diff
@@ -93,7 +93,7 @@ detail::TextureImpl* Load( File& F )
         Succ = F.Read( Buf, InSize );
         if( Succ )
         {
-            detail::TextureImpl::ConvertRGBtoRGBA( ( uint8_t* )( void* )Buf.c_str(), ImgSize, &impl->mData.at( 0 ) );
```
Comment (weanti) [inline src/platform/tgatexture.cpp:96]: The original problem here is, according to coverity, the way Width() and Height() is returned. Maybe that could be changed? I think ImgSize is more intuitive then Buf.size() / channel.
Just an opinion, not a recommendation.

**Comment by weanti** [comment]
Altogether is looks fine.

File: src/platform/tgatexture.cpp (hunk 2 above)
Comment (weanti) [inline src/platform/tgatexture.cpp:96]: Ah disregard my comment, I see no better way for Width() and Height().
So the question is: a) can we suppress warnings b) is it worth fixing it's a bit less readable?

Comment (MrPepperoni) [inline src/platform/tgatexture.cpp:96]: Don't know, tbh. suppressing is possible probably. Gonna try with "validating" the input width/height (<= 2^15)

File: src/audio/audio_player.cpp [hunk 3]
```diff
@@ -113,6 +113,10 @@ AudioPlayer::AudioPlayer()
     , mClosing( false )
     , mNumChannels( 2 )
     , mBuffer( mNumChannels )
+    , mOutputParameters()
+    , mStream( NULL )
+    , mPlaybackBuffer( NULL )
```
Comment (abannerth) [inline src/audio/audio_player.cpp:118]: Why not `nullptr`?

Comment (MrPepperoni) [inline src/audio/audio_player.cpp:118]: Cus i'm oldschool. Or just old. Good point btw.

**Comment by abannerth** [comment]
Since two of us looked at it and weanti already wrote it looks fine, I'm just gonna go ahead and merge it.

**Pull request merged**